zerver/lib/bugdown/__init__.py

```diff
@@ -1638,13 +1692,19 @@ def do_convert(content, message=None, message_realm=None, possible_words=None, s
         else:
             realm_emoji = dict()
 
+        # Team names use same regex as full names.
+        # TODO: Discuss: What will be the behavior when team name is same as
+        # user name?
+        team_name_info = get_team_name_info(message_realm.id, full_names)
```

Comment:
We might want to have a slightly different syntax for team mentions than user mentions.  This would push the problem out to the client of disambiguating team vs. user in typeahead.  Also, we want to be a little careful about performance here.  If we can scan the message content for potential team mentions before full rendering, we can often skip the get_team_name_info() query.

Reply:
Cool cool. Makes sense.

Reply:
I think our current thinking is we'll do a syntax like `@*support team*`.

Reply:
OK.